zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Note.java

```diff
@@ -754,6 +766,20 @@ public void persist(AuthenticationInfo subject) throws IOException {
     repo.save(this, subject);
   }
 
+  public void rename(AuthenticationInfo subject) throws IOException {
+    Preconditions.checkNotNull(subject, "AuthenticationInfo should not be null");
+    stopDelayedPersistTimer();
+    snapshotAngularObjectRegistry(subject.getUser());
+    index.updateIndexDoc(this);
+    
+    FileInfo currentFile = this.getFileInfo().copy();
+    FileInfo newFile = FileInfo.createInstance();
+    // assuming in the same folder
+    newFile.setFolder(currentFile.getFolder());
+    newFile.setFile(Util.convertTitleToFilename(this.getName()));
+    this.setFileInfo(repo.rename(currentFile, newFile, subject));
```

Comment:
`FileInfo` is used to store current file path under `notebook` dir of the note. Thus, `oldName` represents old (current) path of note before rename, and `newFile` represents where to move the note. This approach is helpful for connecting `folder` structure in GUI with physical folder structure, in that case we could use it if folder was create/changed on front, we would call it with `a/b/noteA.zpln` -> `c/d/noteA.zpln`